c7n/resources/elasticsearch.py

```diff
@@ -622,3 +624,114 @@ def process(self, resources):
         for r in resources:
             client.update_elasticsearch_domain_config(DomainName=r['DomainName'],
                 DomainEndpointOptions={'EnforceHTTPS': True, 'TLSSecurityPolicy': tls_value})
+
+
+@ElasticSearchDomain.action_registry.register('enable-auditlog')
+class EnableAuditLog(Action):
+
+    """Action to enable audit logs on a domain endpoint
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: enable-auditlog
+                resource: elasticsearch
+                filters:
+                  - type: value
+                    key: 'LogPublishingOptions.AUDIT_LOGS.Enabled'
+                    op: eq
+                    value: false
+                actions:
+                  - type: enable-auditlog
+                    state: True
+                    loggroup_prefix: "/aws/es/domains"
+
+    """
+
+    schema = type_schema(
+        'enable-auditlog',
+        state={'type': 'boolean'},
+        loggroup_prefix={'type': 'string'},
+        required=['state'])
+
+    statement = {
+        "Sid": "OpenSearchCloudWatchLogs",
+        "Effect": "Allow",
+        "Principal": {"Service": ["es.amazonaws.com"]},
+        "Action": ["logs:PutLogEvents", "logs:CreateLogStream"],
+        "Resource": None}
+
+    permissions = (
+        'es:UpdateElasticsearchDomainConfig',
+        'es:ListDomainNames',
+        'logs:DescribeLogGroups',
+        'logs:CreateLogGroup',
+        'logs:PutResourcePolicy')
+
+    def get_loggroup_arn(self, domain, log_prefix=None):
+        if log_prefix:
+            log_group_name = "%s/%s/audit-logs" % (log_prefix, domain)
+        else:
+            log_group_name = "/aws/OpenSearchService/domains/%s/audit-logs" % (domain)
+        log_group_arn = "arn:aws:logs:{}:{}:log-group:{}:*".format(
+            self.manager.region, self.manager.account_id, log_group_name)
+        return log_group_arn
+
+    def merge_dict(self, d1, d2):
+        merged_dict = defaultdict(list)
+
+        for d in (d1, d2):
+            for key, value in d.items():
+                if key == 'Resource':
+                    if isinstance(value, list):
+                        merged_dict[key].extend(value)
+                    else:
+                        merged_dict[key].append(value)
+                else:
+                    merged_dict[key] = value
+
+        return dict(merged_dict)
+
+    def set_permissions(self, log_group_arn):
+        statement = dict(self.statement)
+        statement['Resource'] = [log_group_arn]
+        client = local_session(
+            self.manager.session_factory).client('logs')
+
+        try:
+            client.create_log_group(logGroupName=log_group_arn.split(":")[-2])
+        except client.exceptions.ResourceAlreadyExistsException:
+            self.log.warning("Log group already exists ...")
```

Comment:
given this is a normal handled case, its not clear if a warning level is appropriate, or really any logging?

Reply:
fixed